app_flutter/lib/widgets/task_grid.dart

```diff
@@ -148,15 +153,21 @@ class _TaskGridState extends State<TaskGrid> {
           final QualifiedTask qualifiedTask = QualifiedTask.fromTask(task);
           if (commitCount <= 25) {
             double score = 0.0;
+            // Reruns take up extra infra capacity and should be prioritized.
             if (task.attempts > 1) {
-              score += 1.0;
+              score += 0.3;
```

Comment:
Do we want to distinguish success and failure for the case of attempts>1, like `failure` with a slightly higher score?

Reply:
That's how it works as it also adds the weights for those statuses 😅 

Do you think this would be more readable if I generated a bunch of different task categories and hard coded their weights instead of having these modifiers? In this case, `failure-rerun` and `success-rerun`?

Instead of all these extra checks, the weighting algorithm would just be `score = _statusScores[status];`. It would create a much larger setup to define all these, but the weight would be clearly defined.

Reply:
Ah.. sorry missed the status part. I think this is fine. A doc may be good to explicitly list the order of different cases.

Reply:
> Do you think this would be more readable if I generated a bunch of different task categories and hard coded their weights instead of having these modifiers? In this case, `failure-rerun` and `success-rerun`?

Yes, that would make this much more readable.  Right now I'm not really sure what the magic numbers are (why 0.3 here?)

Reply:
Added!